coderd/healthcheck/provisioner.go

```diff
@@ -0,0 +1,141 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/coder/coder/v2/buildinfo"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/healthcheck/health"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/coderd/util/apiversion"
+	"github.com/coder/coder/v2/coderd/util/ptr"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// @typescript-generate ProvisionerDaemonsReport
+type ProvisionerDaemonsReport struct {
+	Severity  health.Severity  `json:"severity"`
+	Warnings  []health.Message `json:"warnings"`
+	Dismissed bool             `json:"dismissed"`
+	Error     *string          `json:"error"`
+
+	ProvisionerDaemons []codersdk.ProvisionerDaemon `json:"provisioner_daemons"`
+}
+
+type ProvisionerDaemonsReportOptions struct {
+	CurrentVersion    string
+	CurrentAPIVersion *apiversion.APIVersion
+
+	// ProvisionerDaemonsFn is a function that returns ProvisionerDaemons.
+	// Satisfied by database.Store.ProvisionerDaemons
+	ProvisionerDaemonsFn func(context.Context) ([]database.ProvisionerDaemon, error)
```

Comment:
I think a clearer pattern is to define an `interface` that includes `ProvisionerDaemons()`.  Then in product code you can pass the database.Store and in test you can pass your own faked thing.

Reply:
I think either is fine when it's only one method, but if there's a need for more, I'd recommend interface too.

I think the `Fn` suffix could be dropped in either case, though. It's redundant since it's already part of the type.